simple.js

```diff
@@ -1396,6 +1396,11 @@ define(function(require, exports, module) {
 
         var loaded = false;
         function load() {
+
+            // remove trailing ? or & (e.g., when left after restting setting)
+            if (history.pushState && /\?*&*$/.test(window.location.href))
```

Comment:
How did the `?` and `&` end up in the URL, though? And each `*` can probably be `?`?

And does `pushState` support replacing the current state? Not sure we want to keep the `?&` version in one's history?

Reply:
To reset the setting we pass `?reset=project|state|user`, and Cloud9's settings plugin removes the parameter from the URL but doesn't remove the `?` (or `&`) if either exists per https://github.com/c9/core/blob/master/plugins/c9.core/settings.js#L41.

> And each `*` can probably be `?`?

I suppose we could but if a user already has more than one `&` for example (as a result of multiple previous resets), they'd have to either remove them manually or reload the workspace multiple times. I don't think it'd harm to use `*`?

> And does pushState support replacing the current state? Not sure we want to keep the ?& version in one's history?

Not sure I get that. Mind elaborating more please?

Reply:
Oh, never mind. I think I understand your question now. Checking ...

Reply:
Changed to `replaceState` instead per fd2c953.